### PR TITLE
WIP: box.rb as default filename

### PR DESCRIPTION
**Pull request by cmaujean**
Live PR for feedback on approach

* still needs help text and tests

Signed-off-by: Christopher Maujean <christopher@maujean.org>

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -32,7 +32,10 @@ var (
 	// Copyright is the copyright, generated automatically for each year.
 	Copyright = fmt.Sprintf("(C) %d %s - Licensed under MIT license", time.Now().Year(), Author)
 	// UsageText is the description of how to use the program.
-	UsageText = "box [options] filename"
+	UsageText = "box [options] <filename>"
+
+	// canconical filename
+	defaultFile = "box.rb"
```
Comment (erikh) [inline main.go]: make this a part of the action please. no globals.

File: main.go [hunk 2]
```diff
@@ -258,3 +262,11 @@ func mkBuilder(cancel context.CancelFunc, buildConfig builder.BuildConfig) (*bui
 	signal.Handler.AddRunner(buildConfig.Runner)
 	return b, nil
 }
+
+func validateCanonicalFile() {
```
Comment (erikh) [inline main.go]: this should be validated inside the builder itself, not here i think. If you remove this check and just leave the file alone without checking it, the file will fail to load anyway and cause a different error. I think this is ok.

File: cli-tests/basic_test.go [hunk 3]
```diff
@@ -94,3 +96,29 @@ func (s *cliSuite) TestVersion(c *C) {
 	cmd.Run()
 	c.Assert(strings.Contains(cmd.Stdout(), "box version"), Equals, true)
 }
+
+func (s *cliSuite) TestCononicalFile(c *C) {
```
Comment (erikh) [inline cli-tests/basic_test.go]: `Canonical`

File: main.go [hunk 4]
```diff
@@ -33,7 +33,10 @@ var (
 	// Copyright is the copyright, generated automatically for each year.
 	Copyright = fmt.Sprintf("(C) %d %s - Licensed under MIT license", time.Now().Year(), Author)
 	// UsageText is the description of how to use the program.
-	UsageText = "box [options] filename"
+	UsageText = "box [options] <filename>"
```
Comment (erikh) [inline main.go]: I've always interpreted the `<>` to imply that the argument is required. Maybe in `[]` instead?

**Comment by cmaujean** [comment]
closing for new non-wip pr

**Pull request closed without merging**